File: Python/Problems/Problems/problem1979.py
import heapq
import math
import queue
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter, deque

from .problem import Problem

class Problem1979(Problem):
    def test(self) -> bool:
        nums = [2, 5, 6, 9, 10]

        expected = 2

        result = self.findGCD(nums)

        print(result)

        return result == expected

    def findGCD(self, nums: List[int]) -> int:
        mx = 0
        mn = 1001
        for num in nums:
            mn = min(mn, num)
            mx = max(mx, num)

        while mn != 0:
            mx %= mn
            (mx, mn) = (mn, mx)

        return mx

File: C++/Problems/Problems/problem_1979.cpp
#include "problem_1979.h"

bool problem_1979::test() {
    std::vector nums { 2, 5, 6, 9, 10 };

    int expected = 2;

    auto result = findGCD(nums);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_1979::findGCD(std::vector<int> &nums) {
    int mx = 0;
    int mn = 1001;
    for (auto num : nums)
    {
        mn = std::min(mn, num);
        mx = std::max(mx, num);
    }

    while (mn != 0)
    {
        mx %= mn;
        std::swap(mx, mn);
    }

    return mx;
}
